bin/dynamic_challenge_score.py

```diff
@@ -0,0 +1,437 @@
+        "Lorenz96": {"function": pde_forecast, "params": {"k": 20, "modes": 30}},
+        "Kolmogorov": {
+            "function": pde_forecast_2d,
+            "params": {"k": 20, "modes": 30, "nf": 128},
+        },
+    }
+
+    if system in system_to_forecast:
+        forecast_func = system_to_forecast[system]["function"]
+        forecast_params = system_to_forecast[system]["params"]
+        scores = forecast_func(truth, prediction, **forecast_params)
+        return list(scores)
+    else:
+        return []
+
+
+def reconstruction(truth: np.ndarray, prediction: np.ndarray) -> float:
+    """Produce reconstruction fit score.
+
+    Arguments:
+        truth: comparison data
+        prediction: predicted data
+
+    Returns:
+        e1: reconstruction fit score
+    """
+    est = np.linalg.norm(truth - prediction, 2) / np.linalg.norm(truth, 2)
+
+    e1 = 100 * (1 - est)
+
+    return e1
+
+
+# TODO: Not final, update once organizers confirm all inputs and metrics
+def calculate_all_scores(
+    groundtruth_path: str, predictions_path: str, evaluation_id: str
+) -> dict:
+    """Calculate scores across all testing datasets.
+
+    Arguments:
+        groundtruth_path: path to the groundtruth folder
+        predictions_path: path to the predictions file
+        evaluation_id: id of the evaluation queue
+
+    Returns:
+        score_result: dictionary containing scores
+    """
+    score_result = {}
+    task_mapping = {
+        "9615379": [("X1", "forecast", ["stf_E1", "ltf_E2"], [0, 1])],  # Task1
+        "9615532": [  # Task2
+            ("X2", "reconstruction", ["recon_E3"], [0]),
+            ("X3", "forecast", ["ltf_E4"], [1]),
+            ("X4", "reconstruction", ["recon_E5"], [0]),
+            ("X5", "forecast", ["ltf_E6"], [1]),
+        ],
+        "9615534": [("X6", "forecast", ["stf_E7", "ltf_E8"], [0, 1])],  # Task3
+        "9615535": [  # Task4
+            ("X7", "forecast", ["stf_E9", "ltf_E10"], [0, 1]),
+            ("X8", "reconstruction", ["recon_E11"], [0]),
+            ("X9", "reconstruction", ["recon_E12"], [0]),
+        ],
+    }
+
+    # get mapping of inputs and outs for specific task
+    task_info = task_mapping.get(evaluation_id)
+
+    # get unique systems
+    pred_files = os.listdir(predictions_path)
+    pred_systems = list(set(f.split("_")[0] for f in pred_files))
+    true_systems = [
+        "doublependulum",
+        "Lorenz",
+        "Rossler",
+        "Lorenz96",
+        "KS",
+        "Kolmogorov",
+    ]
+    unique_systems = list(set(true_systems) & set(pred_systems))
+
+    for system in unique_systems:
+        for prefix, score_metric, score_keys, score_indices in task_info:
+            truth_path = os.path.join(
+                groundtruth_path, f"Test_{system}/{prefix}test.npy"
+            )
+            pred_path = os.path.join(
+                predictions_path, f"{system}_{prefix}prediction.npy"
+            )
+
```

Comment:
This function is outdated - see above comment.
There is an extra validation to score only on provided prediction files